src/software/thunderscope/gl/graphics/gl_heatmap.py

```diff
@@ -0,0 +1,123 @@
+from pyqtgraph.opengl import *
+from pyqtgraph.opengl.GLGraphicsItem import GLGraphicsItem
+from typing import Optional, Callable, Tuple
+
+import numpy as np
+import numpy.typing as npt
+
+ColorMap = Callable[[float], Tuple[float, float, float, float]]
+
+
+class GLHeatmap(GLMeshItem):
+    """
+    Displays a heatmap on the cartesian plane (i.e. x-y plane)
+    """
+
+    def __init__(
+        self, parent_item: Optional[GLGraphicsItem] = None, color_map: ColorMap = None,
+    ) -> None:
+        """Initialize the GLHeatmap
+        
+        :param parent_item: The parent item of the graphic
+        :param color_map: Function to map data values to color space.
+                          Returns a 4-tuple representing the RGBF color for the 
+                          input data value (note that 0 <= RGBF value <= 1).
+                          The default color_map will map data values in the
+                          range [0, 1] to a grayscale color space. 
```

Comment:
nit: Is it RGBF or RGBA?

Reply:
actually I guess it should be rgbaF where the F stands for float (I think). Weird Qt thing where rgba colors can be represented as 4 floats each from 0 to 1, instead of 4 ints 0 to 255

Reply:
Changed this; now using the actual ColorMap provided by pyqtgraph